doc/guide/troubleshooting.rst

```diff
@@ -0,0 +1,448 @@
+
+At this time, one or more :option:`prolog-start` events may be posted to
+the eventlog by jobtap plugins. These events prevent the start request
+from the job manager to the job execution system until a corresponding
+:option:`prolog-finish` event is emitted.
+
+Once all :option:`prolog-finish` events have been posted, the start request
+is sent and a :option:`start` event is posted to the job eventlog when
+the job execution system has launched all the job shells for the job.
+
+.. code-block:: console
+
+  # flux job eventlog -H fmctRr2YQ8f
+  [snip]
+  [  +0.122930] prolog-start description="job-manager.prolog"
+  [  +4.288898] prolog-finish description="job-manager.prolog" status=0
+  [  +4.328720] start
+
+The :option:`job-manager.prolog` is managed by the :command:`perilog.so`
+jobtap plugin, and is responsible for invoking the per-node job prolog
+when configured (see :man5:`flux-config-job-manager`). Other plugins may
+post :option:`prolog-start` events to prevent the job from starting while
+they perform some kind of job prolog action.
+
+If the :option:`prolog-finish` event for the :option:`job-manager.prolog`
+is not posted in a timely manner, debug information can be obtained
+directly from the :command:`perilog.so` plugin:
+
+.. code-block:: console
+
+  # flux jobtap query perilog.so | jq .procs
+  {
+    "ƒ22EmWZk1mkb": {
+      "name": "prolog",
+      "state": "running",
+      "total": 4,
+      "active": 1,
+      "active_ranks": "0",
+      "remaining_time": 55
+    }
+  }
+
+The above shows that a prolog for job :option:`ƒ22EmWZk1mkb` is currently
+running. It was executed on a :option:`total` of 4 broker ranks, and still
+currently :option:`active` on 1 rank, rank 0. The prolog will time out in
+55 seconds.
+
+In this case there may be an issue on rank 0, since the prolog completed
+on the other involved ranks already. Log into the host associated with broker
+rank 0 and list the process tree of the :command:`flux-prolog@JOBID` unit
+to see the process tree of the prolog:
+
+.. code-block:: console
+
+  # flux overlay lookup 0
+  host1
+  # ssh host1
+  # systemd-cgls -u flux-prolog@f22EqDxY5rrK.service
+  Unit flux-prolog@f22EqDxY5rrK.service
+  ├─2056298 /bin/sh /etc/flux/system/prolog
+  ├─2056301 /bin/sh /etc/flux/system/prolog.d/doit.sh
+  └─2056302 hang
+
+Since the prolog and epilog are executed as ephemeral systemd units,
+the output from these scripts can be obtained from
+:command:`journalctl -u flux-prolog@*`.
+
+*************
+Job Execution
+*************
+
+Once all :option:`prolog-start` events have a corresponding
+:option:`prolog-finish` event, the job manager sends a start request to
+the job execution system. The :command:`job-exec` module launches job
+shells (via the IMP on a multi-user system). Once all shells have started
+a :command:`start` event is posted to the main job eventlog.
+
+When the start event is delayed, the :command:`job-exec` module can
+be queried for the job to get some detail:
```

Comment:
most places you don't `:command:`MODULE NAME``, intentionally wanted to do it for `job-exec`?  I'm assuming just a formatting error.

Reply:
It's actually a good question what we should use here. I know the bare minimum about sphinx/rst, I just know I don't want red text like double-backticks give you. Perhaps there's a way we can make our own "domains" for modules, job states, etc. But for now I'll make it consistent and fix this case (you're right it was just an accident in this case)

Reply:
I settled on using `:command:` only for commands (and DSOs) and `:option:` for everything else. If this was incorrect, it should be easy to come along and fix it later.